src/lib/structures/expr.mli

```diff
@@ -168,14 +168,14 @@ val is_fresh : t -> bool
 val is_fresh_skolem : t -> bool
 val is_int : t -> bool
 val is_real : t -> bool
-val type_info : t -> Ty.t
-
+val [@inline always] type_info : t -> Ty.t
```

Comment:
these annotations are useless in `.mli` files afaik.

Reply:
Does that mean if the mli does specify this inlining but the ml does not, the function will not be inlined?

Reply:
If I'm correct, in an `.mli` file, the attribute is simply ignored (i.e. I don't think the inlining code ever looks at attributes on the type declaration, or even has access to the type declaration).

Reply:
You should remove the attributes from the mli files: they are literally ignored/unused by the compiler, and they are surprising/quite distracting when reading an `.mli` file.

Reply:
I disagree with you here; I find useful to see directly when a function will be inlined without having to see the implementation.

Reply:
In that case, you can always mention it in the doc comments.
To expand a bit, here are some reasons why I think it's bad to have such attributes in .mli files:
- it is currently silently ignored by the compiler, but there are upcoming patches that would emit warnings on ignored attributes, warnings that dune would likely treat as errors during dev.
- I'm not sure why you  would want to know whether a function will be inlined. The only situation where it is really important/critical is when the function occurs in a hot path, and when the function body is really small (with a size/runtime comparable to that of a function call). In such cases, if you're interested in performances, you already need to know what the implementation of the function does (or at the very least, its complexity), so you need knowledge of the function's body, which defeats the purpose of having the annotation in the mli file.
- inline annotations/attributes are only read/acted upon by flambda, so most of the time (as long as your switch did not explicitly enable flambda), all such annotations are ignored, so you cannot and should not rely on functions being inlined, and thus having that info in the mli file seems... ineffective.
- very small functions such as these ones, will already be unconditionally inlined, even with closure; and with flambda, the decision of whether to inline is made at the call-sites rather than at the definition site, so again, annotating the interface doesn't make much sense.

Overall, I'd say that the goal of being a bit more aware of performance (which I assume is why you find it useful to know whether a function will be inlined), would be more served by providing an estimation of the complexity (in terms of run time/memory) in the doc comment of the function, and rely on the fact that very simple/trivial functions will 1) either be naturally inlined by the compiler without any annotation (because the ocaml compiler is not that bad), or 2) will have its definition/implementation annotated to guide the flambda inliner, if it is pertinent (which depends on the implementation, not the interface).

Reply:
* If in the future dune will emit warnings on ignored attributes, then it is a good idea to write them in the `.mli` as a safeguard when it comes to update the code don't you think?
* It is just a question of documentation and clarification. The annotation was here in the first place actually, and the more information I can get in the .mli the less I am likely to make wrong assumptions in general.
* I could as well remove all `.mli`s before compiling and I would still be able to compile the project the exact same way.
* I did not know that ^^

I will save us some efforts and remove this annotation anyway.